src/Common/tests/System/AssertExtensions.cs

```diff
@@ -104,5 +104,18 @@ public static void ThrowsAny(Type firstExceptionType, Type secondExceptionType,
         {
            ThrowsAny(typeof(TFirstExceptionType), typeof(TSecondExceptionType), action);
         }
+
+        public static void ThrowsIf<T>(Action action, bool condition)
```

Comment:
I suggest the arguments be swapped here to have the condition be first.  That better aligns with APIs like Debug.WriteLineIf.